Lib9c/Model/Item/ShopItem.cs

```diff
@@ -20,33 +21,53 @@ public class ShopItem
         public readonly FungibleAssetValue Price;
         public readonly ItemUsable ItemUsable;
         public readonly Costume Costume;
+        public readonly long ExpiredBlockIndex;
 
         public ShopItem(Address sellerAgentAddress,
             Address sellerAvatarAddress,
             Guid productId,
             FungibleAssetValue price,
-            ItemUsable itemUsable)
+            ItemUsable itemUsable) : this(sellerAgentAddress, sellerAvatarAddress, productId, price, 0, itemUsable)
         {
-            SellerAgentAddress = sellerAgentAddress;
-            SellerAvatarAddress = sellerAvatarAddress;
-            ProductId = productId;
-            Price = price;
-            ItemUsable = itemUsable;
-            Costume = null;
         }
 
         public ShopItem(Address sellerAgentAddress,
             Address sellerAvatarAddress,
             Guid productId,
             FungibleAssetValue price,
-            Costume costume)
+            Costume costume) : this(sellerAgentAddress, sellerAvatarAddress, productId, price, 0, costume)
+        {
+        }
+
+        public ShopItem(
+            Address sellerAgentAddress,
+            Address sellerAvatarAddress,
+            Guid productId,
+            FungibleAssetValue price,
+            long expiredBlockIndex,
+            INonFungibleItem nonFungibleItem
+        )
         {
             SellerAgentAddress = sellerAgentAddress;
             SellerAvatarAddress = sellerAvatarAddress;
             ProductId = productId;
             Price = price;
-            ItemUsable = null;
-            Costume = costume;
+            if (expiredBlockIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException($"ExpiredBlockIndex must be 0 or more, but {expiredBlockIndex}");
```

Comment:
How about using `nameof` keyword?

```suggestion
                throw new ArgumentOutOfRangeException($"{nameof(ExpiredBlockIndex)} must be 0 or more, but {expiredBlockIndex}");
```